dcp-0003/dcp-0003.mediawiki

```diff
@@ -0,0 +1,837 @@
+<code>CHECKSEQUENCEVERIFY</code> opcode in this proposal, it is possible to
+modify the example contract slightly such that the timeout is relative to the
+time the contract is actually mined into the chain as follows:
+
+    OP_IF
+      OP_SHA256 <hash of secret> OP_EQUALVERIFY           // Require disclosure of secret to redeem
+      OP_DUP OP_HASH160 <counterparty2 public key hash>   // Require signature from counterparty 2
+    OP_ELSE
+      <locktime> OP_CHECKSEQUENCEVERIFY OP_DROP           // Prevent redemption until specified relative timeout after funding
+      OP_DUP OP_HASH160 <counterparty1 public key hash>   // Require signature from counterparty 1
+    OP_ENDIF
+    OP_EQUALVERIFY OP_CHECKSIG                            // Ensure signature is valid
+
+This is useful because it allows these contracts to be negotiated off-chain
+without imposing an absolute deadline by which they have to be recreated with a
+timeout further in the future.  This is the case because, due to the relative
+timeout, the clock does not start ticking until the transaction is actually
+mined into the chain.
+
+===Bidirectional Payment Channels===
+
+A fundamental requirement of the Lightning Network (LN) is what is known as a
+bidirectional payment channel.  In essence, a bidirectional payment channel is a
+series of off-chain transactions that are constructed in a way that allows them
+to take advantage of the security of normal on-chain transactions while
+simultaneously providing several benefits.
+
+At a high level, this is accomplished by creating an on-chain anchor transaction
+that requires signatures from both counterparties to redeem a series of
+time-locked revocable commitment transactions that spend from that anchor
+transaction.  Each party initially creates a commitment transaction that refunds
+their respective individual amounts they used to fund the contract, however,
+they do not broadcast them to the network.  Instead, as they conduct business,
+they create new commitment transactions which alter how the funds are split
+between them.  Since the anchor transaction can only be spent once, and due to
+the restrictions enforced by the underlying smart contract, only the most recent
+commitment transaction is valid.  Consequently, the payment channel can be
+trustlessly closed at any time by either party by simply broadcasting the
+most recent commitment transaction to the network for inclusion in the
+blockchain.
+
+Since the commitment transactions are time locked, using absolute times would
+effectively place a lifetime on the channel.  This is the case because it
+would be necessary to close and reopen the channel any time the absolute timeout
+was approaching in order to prevent the counterparty from being able to steal
+funds.  This also implies that it would be necessary to use fairly long timeouts
+with absolute time locks in order to limit the frequency of channel churn.  Long
```

Comment:
Not clear to me what is meant by channel churn.  Creation and closing?

Reply:
Correct.  I changed it to say: "limit the frequency of opening and closing
channels".